### PR TITLE
feat: add notification center spotlight

### PR DESCRIPTION
**Pull request by kathaypacific**
### Description

This PR:
- add a new spotlight component that covers the whole screen. since it's not possible to make a "hole" in the modal itself, i'm faking the spotlight by drawing the notification bell on top of the spotlight in the same place.
- add new redux state that tracks whether to show the spotlight. this defaults to false, and is only set to true with redux migration / app upgrade.
- this is a one time use component so i didn't try to optimise too much, there's a bit of clunky coordination to get all the absolutely positioned components working together.

### Test plan

Unit tests. 

Manually: 
- as a new install, i should not see the spotlight
- as an upgrading user, i should see the spotlight
- the styling should be correct on iOS / Android / on devices with and without top notches

https://github.com/valora-inc/wallet/assets/20150449/d256d006-4964-47a6-9cf8-277a3d8ff466

Simulator Screenshot - iPhone 14 Pro - 2023-09-15 at 17 20 02

### Related issues

- Fixes RET-802

### Backwards compatibility

Y

### CODE REVIEW

**Review by bakoushin — APPROVED** [review]
LGTM!

**Review by jeanregisser — APPROVED** [review]
Awesome! 🌟 
Nice simple solution.

Just double checking, since I didn't see a screenshot, the spotlight position is also correct on Android?

File: src/home/NotificationBellSpotlight.tsx [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+interface Props {
+  isVisible: boolean
+}
+
+const VERTICAL_TOP_BAR_OFFSET = 4
+const HORIZONTAL_TOP_BAR_OFFSET = 4
+const SPOTLIGHT_SIZE = 48
+const ARROW_SIZE = 8
+
+export default function NotificationBellSpotlight({ isVisible }: Props) {
```
[the remark this answers is not in the document]
Comment (kathaypacific) [reply src/home/NotificationBellSpotlight.tsx:29]: sure, added!

**Comment by kathaypacific** [comment]
> Awesome! 🌟 Nice simple solution.
> 
> Just double checking, since I didn't see a screenshot, the spotlight position is also correct on Android?

yes i checked on android - the video in the description is on android. i also checked android with notch just now and looks fine :) 

Screenshot_20230919_091717_Alfajores (dev)

**Pull request merged**